cmd/kubeadm/app/cmd/init.go

```diff
@@ -249,6 +249,22 @@ func NewInit(cfgPath string, cfg *kubeadmapi.MasterConfiguration, ignorePrefligh
 		return nil, err
 	}
 
+	// if the cluster should be created with features.HighAvailability enabled
+	if !features.Enabled(cfg.FeatureGates, features.HighAvailability) {
+
+		// Checks if the cluster meets the requirements for kubeadm HighAvailability / current version
+
+		// 1. Checks if the cluster uses an external etcd
+		if len(cfg.Etcd.Endpoints) == 0 {
+			return nil, fmt.Errorf("a cluster with '--feature-gates=HighAvailability=true' requires an external etcd")
+		}
+
+		// 2. Checks if the cluster uses an external load balancer
+		// TODO
+		// ** Note for reviewers:
+		// ** How to check this? Check if advertise addres != any IP address of the machine's newtwork interfaces?
```

Comment:
yes.

Reply:
IMO This is actually the most tricky part of the PR so far.
Let's discuss this in the next meeting

The short answers is that I don't like the idea of the advertise address == master Ip because it makes things much more complicated (I will argument this in the meeting). 
My suggestion for fixing the AWS problem is to accept a DNS name as advertise address in case of --feature-gates=HighAvailability. WDYT?

Note: 
In this PR I'm limited to a single workflow on the machine where kubeadm is executed, while in #546 we are discussing solutions where you can use automation tools and basically move back and forth from different machines. 
However, it will be really interesting to get people working on #546 checking if the current PR simplifies their automations scripts...